typescript/packages/wallets/crossmint/src/wallets/SolanaSmartWalletClient.ts

```
@@ -0,0 +1,140 @@
+    if (linkedUser) {
+        if (linkedUser.email) return `email:${linkedUser.email}:solana-smart-wallet`;
+        if (linkedUser.phone) return `phone:${linkedUser.phone}:solana-smart-wallet`;
+        if (linkedUser.userId) return `userId:${linkedUser.userId}:solana-smart-wallet`;
+    }
+    if (!address) throw new Error("Either address or linkedUser must be provided");
+    return address;
+}
+
+export class SolanaSmartWalletClient extends SolanaWalletClient {
+    readonly #locator: string;
+    readonly #api: CrossmintWalletsAPI;
+    readonly #address: string;
+
+    constructor(api: CrossmintWalletsAPI, address: string, options: SolanaSmartWalletOptions) {
+        super({ connection: options.connection });
+        this.#api = api;
+        this.#address = address;
+        this.#locator = getLocator(options.address, options.linkedUser);
+    }
+
+    getAddress() {
+        return this.#address;
+    }
+
+    async signMessage(message: string) {
+        try {
+            const { id: signatureId } = await this.#api.signMessageForCustodialWallet(this.#locator, message);
+            while (true) {
+                const latestSignature = await this.#api.checkSignatureStatus(signatureId, this.#address);
+                if (latestSignature.status === "success") {
+                    if (!latestSignature.outputSignature) {
+                        throw new Error("Signature is undefined");
+                    }
+                    return {
+                        signature: latestSignature.outputSignature,
+                    };
+                }
+                if (latestSignature.status === "failed") {
+                    throw new Error("Signature failed");
+                }
+                await new Promise((resolve) => setTimeout(resolve, 2000));
+            }
+        } catch (error) {
+            throw new Error(`Failed to sign message: ${error}`);
+        }
+    }
```

Signatures are not (and probably will never be) available for smart wallets! The reason is that in Solana a smart wallet is an off-curve address and we do not have any private key that we could use to sign. This method should throw an error